global/scripts/mppmodecheck.shlib

```diff
@@ -0,0 +1,231 @@
+# Functions for checking compliance with selected MPP mode for HCP Pipelines
+
+# --- Import log.shlib 
+
+if [[ "$(type -t log_Err_Abort)" != "function" ]]
+then
+    source "$HCPPIPEDIR/global/scripts/log.shlib" "$@"
+fi
+
+
+# --- Functions to check that a valid pipeline name was passed
+
+check_pipeline_name()
+{
+    for valid_pipeline in PreFreeSurfer FreeSurfer FreeSurfer53HCP PostFreeSurfer fMRIVolume fMRISurface Difussion
+    do
+        if [ "${valid_pipeline}" = "${pipeline}" ]; then
+            ispipeline="true"
+        fi
+    done
+}
+
+# --- Functions for pipeline specific checks
+
+check_mpp_PreFreeSurfer()
```

Comment:
I was thinking of leaving the logic of what is or is not hcp-compliant in the pipelines, and this file would just contain the standard log, warning, and error messages.  That way, this file wouldn't need to know any details about the pipelines.

Reply:
That was what I interpreted your proposal as (just a simple new function added to the log.shlib library), and I like the idea of not needing to go to the `check_mpp_` functions to see the relevant logic.

Reply:
I was not sure about the initial suggestion. Then it made sense that all the criteria for what is and what is not HCPStyleData or LegacyStyleData are in one file, so that it is easy to harmonize the checks across multiple scripts. 

I can restructure the code. But before I do, let's discuss pro's and con's. @mharms @glasserm, do you have a preference?

Reply:
The design rule I am thinking of is that the fewer files know the internal details of other files, the better (if we swapped out the argument parsing code in a pipeline, variable names could change, breaking the current sourced functions that use these internal details).  As long as we need separate code per pipeline for checking each pipeline's compliance, that code should be in the same file as the pipeline it checks (especially if it uses internal variable names).

My suggestion to consolidate the long identical messages was simply that if we wanted to tweak the wording, editing the identical wording in 4+ files would have been inefficient and error-prone.

Reply:
I agree with @coalsont's concern here, and prefer his suggested approach.

Reply:
implemented